src/Controller/Organization/SearchController.php

```diff
@@ -4,20 +4,28 @@
 
 namespace App\Controller\Organization;
 
+use App\Entity\Organization;
 use App\Repository\CommissionableAssetRepository;
+use App\Repository\OrganizationRepository;
 use App\Repository\UserRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 
 /**
  * @Route("/search", name="app_organization_search", methods={"GET"})
  */
 final class SearchController extends AbstractController
 {
-    public function __invoke(Request $request, UserRepository $userRepository, CommissionableAssetRepository $commissionableAssetRepository): Response
+    public function __invoke(Request $request, UserRepository $userRepository, CommissionableAssetRepository $commissionableAssetRepository, OrganizationRepository $organizationRepository): Response
     {
+        $organization = $this->getUser();
+        if (!$organization instanceof Organization) {
+            throw new AccessDeniedException();
```

Comment:
Should be handled in security.yaml file

Reply:
we could need it for the user entity typehint. Maybe we can create a trait with `getOrganizationUser(): Organization` and `getUserUser(): User` ?

Reply:
If it's just for the typehint, it could be solved like:
```php
/** @var Organization $organization */
$organization = $this->getUser();
```

The AccessDeniedException should be handled by the security.yml, as most of access control securities of a Symfony project

Reply:
I'm not a big fan of phpdoc typehint, but I'm okay for using the security component